library/src/main/java/com/crossbowffs/remotepreferences/RemotePreferenceProvider.java

```diff
@@ -121,10 +183,16 @@ public boolean onCreate() {
         // is created. This method is called before almost all other code in
         // the app, which ensures that we never miss a preference change.
         Context context = getContext();
-        for (String prefName : mPrefNames) {
-            SharedPreferences prefs = context.getSharedPreferences(prefName, Context.MODE_PRIVATE);
+        Context storageContext = null;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N)
+            storageContext = context.createDeviceProtectedStorageContext();
+        else
+            storageContext = context;
+        for (RemotePreferenceFile remotePreferenceFile : mRemotePreferenceFiles) {
```

Comment:
Just `RemotePreferenceFile file` is fine here, reduces some of the line splitting below